modules/http/src/main/java/org/glassfish/grizzly/http/util/HttpDateFormat.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation.
+ * Copyright (c) 2025 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package org.glassfish.grizzly.http.util;
+
+import org.glassfish.grizzly.utils.Charsets;
+
+import java.io.UnsupportedEncodingException;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.glassfish.grizzly.http.util.HttpCodecUtils.toCheckedByteArray;
+
+/**
+ * Utility class to generate HTTP dates.
+ * This class is based on {@link FastHttpDateFormat},
+ * but uses java.time API instead of the old java.util.Date/Calendar and java.text.DateFormat/SimpleDateFormat API.
+ *
+ * @author Bongjae Chang
+ */
+public final class HttpDateFormat {
+
+    private static final String ASCII_CHARSET_NAME = Charsets.ASCII_CHARSET.name();
+
+    private static final int CACHE_SIZE = 1000;
+
+    private static final ZoneId GMT_ZONE_ID = ZoneId.of("GMT");
+
+    // Fri, 03 Oct 2025 04:47:00 GMT
+    private static final DateTimeFormatter FORMATTER =
+            DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.US).withZone(GMT_ZONE_ID);
+
+    private static final DateTimeFormatter[] FORMATTERS = new DateTimeFormatter[]{
+            DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss zzz", Locale.US).withZone(GMT_ZONE_ID),
+            DateTimeFormatter.ofPattern("EEE, dd-MMM-yy HH:mm:ss zzz", Locale.US).withZone(GMT_ZONE_ID),
+            DateTimeFormatter.ofPattern("EEE MMMM d HH:mm:ss yyyy", Locale.US).withZone(GMT_ZONE_ID)};
+
+    /**
+     * Instant on which the currentDate object was generated.
+     */
+    private static volatile long nextGeneration;
+
+    private static final AtomicBoolean isGeneratingNow = new AtomicBoolean();
+
+    /**
+     * Current formatted date as byte[].
+     */
+    private static volatile byte[] currentDateBytes;
```

Comment:
This may be non-volatile

Reply:
> Is this because the `Happens-Before` Guarantee/Relationships affect non-volatile variables?

Yes. Let two variables: non-volatile `a` and volatile `b` and two treads 

```
Thread 1:
a = value1
b = value2 // volatile write

Thread 2:
read b // volatile read
read a
```

Then if `read b` = value2 the `read a` guaranteed to see `value1`.